### PR TITLE
Adds a local s3 implementation using LocalStack

### PR DESCRIPTION
**Pull request by CryptoGraffe**
* addresses issue #450 
* this is hooked up to the laravel storage system, but will be used for future hasura image/file work
* I had to manually do a `docker-compose build` to rebuild the larevel Dockerfile changes that are in the submodule

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -16,6 +16,8 @@ LOCAL_CIRCLE_ADMIN=0xYOUR_DEV_ADDRESS
 # Local Environment
 REACT_APP_API_BASE_URL=http://localhost:9000/api
 REACT_APP_HASURA_URL=http://localhost:8080/v1/graphql
+# local version of s3 for local testing
+REACT_APP_S3_BASE_URL=http://coordinape.s3-website.localhost.localstack.cloud:4566
```
Comment (exrhizo) [inline .env.example:20]: I wonder if this can use a URL, that is just `http://localhost:4566` otherwise not a big deal.

File: docker-compose.yml [hunk 2]
```diff
@@ -31,6 +31,31 @@ services:
...
+  localstack:
+    image: localstack/localstack:latest
+    hostname: s3
+    networks:
+      default:
+        aliases:
+          - s3.localhost.localstack.cloud
+          - s3-website.localhost.localstack.cloud
+          - coordinape.s3.localhost.localstack.cloud
+          - coordinape.s3-website.localhost.localstack.cloud
```
Comment (exrhizo) [inline docker-compose.yml:45]: oh I see, this was done here, is it needed to have so many aliases?

I imagine we could do this with all of the docker images, but probably not worth it, although more readable

File: docker-compose.yml [hunk 3]
```diff
@@ -31,6 +31,31 @@ services:
...
+    environment:
+      - HOSTNAME_EXTERNAL=localstack
+      - AWS_DEFAULT_REGION=us-east-1
+      - EDGE_PORT=4566
```
Comment (exrhizo) [inline docker-compose.yml:49]: having the port as a .env variable seems good, with the other variables

**Review by exrhizo — COMMENTED** [review]
This can be branched off of `main`

File: docker-compose.yml (hunk 2 above)
Comment (CryptoGraffe) [reply docker-compose.yml:45]: I was following examples from LocalStack, i think there is variance in how different versions of s3 refers to the host, older versions use folder paths and newer versions use subdomains. Hence the 2 versions. Then the two other hosts are the webserving hosts, so 4 total.

File: docker-compose.yml (hunk 3 above)
Comment (CryptoGraffe) [reply docker-compose.yml:49]: None of the other ports in this file have vars for them, how do you want to do it? There is a range for localstack but i dont think its necessary, we probably only need 4566.

**Comment by CryptoGraffe** [comment]
> This can be branched off of `main`

Switched to main.

File: docker-compose.yml (hunk 3 above)
Comment (CryptoGraffe) [reply docker-compose.yml:50]: Never mind i see it in main, i'll sort this out

Comment (exrhizo) [reply docker-compose.yml:50]: this is all that is needed to be ready to merge, we can do more for improving env variables in another PR

@levity has feelings about the env as well

Comment (exrhizo) [reply docker-compose.yml:50]: eh, don't want to complexify the env, if it comes up, we can address it then

**Comment by exrhizo** [comment]
I created https://github.com/coordinape/coordinape/pull/512
and will merge that shortly

**Pull request closed without merging**